indexers/social_feed/social_feed.js

```diff
@@ -244,98 +244,115 @@ async function getBlock(block: Block) {
   // Add your code here
   const SOCIAL_DB = "social.near";
 
-  const nearSocialPosts = block
-    .actions()
-    .filter((action) => action.receiverId === SOCIAL_DB)
-    .flatMap((action) =>
-      action.operations
-        .map((operation) => operation["FunctionCall"])
-        .filter((operation) => operation?.methodName === "set")
-        .map((functionCallOperation) => {
-          try {
-            const decodedArgs = base64decode(functionCallOperation.args);
-            return {
-              ...functionCallOperation,
-              args: decodedArgs,
-              receiptId: action.receiptId,
-            };
-          } catch (error) {
-            console.log(
-              "Failed to decode function call args",
-              functionCallOperation,
-              error
-            );
-          }
-        })
-        .filter((functionCall) => {
-          try {
-            const accountId = Object.keys(functionCall.args.data)[0];
-            return (
-              Object.keys(functionCall.args.data[accountId]).includes("post") ||
-              Object.keys(functionCall.args.data[accountId]).includes("index")
-            );
-          } catch (error) {
-            console.log(
-              "Failed to parse decoded function call",
-              functionCall,
-              error
-            );
-          }
-        })
+  let nearSocialPosts = [];
+  try {
+    const actions = block.actions();
+    if (!actions) {
+      console.log("Block has no actions");
+      return;
+    }
+    const contractActions = actions.filter(
+        (action) => action.receiverId === SOCIAL_DB
+    );
+    if (!contractActions) {
+      console.log("Block has no actions");
+      return;
+    }
+    nearSocialPosts = contractActions.flatMap((action) =>
+        action.operations
+            .map((operation) => operation["FunctionCall"])
+            .filter((operation) => operation?.methodName === "set")
+            .map((functionCallOperation) => {
+              try {
+                return {
+                  ...functionCallOperation,
+                  args: base64decode(functionCallOperation.args),
+                  receiptId: action.receiptId, // providing receiptId as we need it
+                };
+              } catch (e) {
+                console.log("Error parsing function call", e);
+              }
+            })
+            .filter((functionCall) => {
+              try {
+                if (
+                    !functionCall ||
+                    !functionCall.args ||
+                    !functionCall.args.data ||
+                    !Object.keys(functionCall.args.data) ||
+                    !Object.keys(functionCall.args.data)[0]
+                ) {
+                  console.log(
+                      "Set operation did not have arg data in expected format"
+                  );
+                  return;
```

Comment:
Does return return false or true? We want to skip the ones that don't work right? Or does this return end execution of the block? 

Also, should we also put the function call in the log statement or no? Just depends on if that's useful for debugging or not.